internal/dag/secret.go

```diff
@@ -122,9 +123,15 @@ func validateCertificate(data []byte) error {
 		if block.Type != "CERTIFICATE" {
 			return fmt.Errorf("unexpected block type '%s'", block.Type)
 		}
-		if _, err := x509.ParseCertificate(block.Bytes); err != nil {
+		cert, err := x509.ParseCertificate(block.Bytes)
+		if err != nil {
 			return err
 		}
+
+		if !hasCommonName(cert) && !hasSubjectAltNames(cert) {
```

Comment:
IIUC, the CN is not required if you have an altname

Reply:
I think the logic I was going for was the cert needs at least one of a CN or an alt name. I think I got that logic right.

Reply:
Erm, yes you did :)

Reply:
| !hasCommonName | !hasSubjectAltName | Result | 
| ------------------- | ---------------------| ------ |
| false                         | false                           | false |
| false                      | true | false |
| true | false | false |
| true | true | true |

I think I got it right, but writing out the truth table hurt.